### PR TITLE
Update RI

### PR DESCRIPTION
**Pull request by v-fearam**
Bicep. 
Enable workload  identity.
SCI update
Be able to  deploy more than one version of the same microservice
Readme Updates.
Nginx update
fix: https://github.com/mspnp/microservices-reference-implementation/issues/234
fix: https://github.com/mspnp/microservices-reference-implementation/issues/233
fix: https://github.com/mspnp/microservices-reference-implementation/issues/232
fix: https://github.com/mspnp/microservices-reference-implementation/issues/229
fix: https://github.com/mspnp/microservices-reference-implementation/issues/228
fix: https://github.com/mspnp/microservices-reference-implementation/issues/226

### CODE REVIEW

**Review by ferantivero — CHANGES_REQUESTED** [review]

File: azuredeploy.bicep [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    nodeResourceGroup: nodeResourceGroupName
+    dnsPrefix: aksClusterName
+    agentPoolProfiles: [
+      {
+        name: 'agentpool'
+        osDiskSizeGB: osDiskSizeGB
+        count: agentCount
+        vmSize: agentVMSize
+        osType: osType
+        mode: 'System'
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply azuredeploy.bicep:114]: I created a task for the future. My idea is as soon as possible have a version working again, and to do improvement later on.

File: charts/delivery/values.yaml [hunk 2]
```diff
@@ -3,7 +3,9 @@ nameOverride: delivery
 replicaCount: 1
 identity:
   clientid:
-  resourceid:
+  serviceAccountName:
+  tenantId:
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/delivery/values.yaml:7]: It is true, good catch, tenantId is not needed on dronescheduler

[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/delivery/values.yaml:7]: yes, It is needed. 
charts\delivery\templates\delivery-secret-provider.yaml
tenantId: {{ .Values.identity.tenantId }}

dronescheduler doesn't have SecretProviderClass

File: charts/ingestion/templates/ingestion-ingress.yaml [hunk 3]
```diff
@@ -15,24 +15,26 @@ kind: Ingress
...
-    kubernetes.io/ingress.class: {{ required "ingress.class is required" .Values.ingress.class | quote }}
+    nginx.ingress.kubernetes.io/ssl-redirect: "true"
+    nginx.ingress.kubernetes.io/use-regex: "true"
     nginx.ingress.kubernetes.io/rewrite-target: /api/deliveryrequests$1
 spec:
+  ingressClassName: {{ required "ingress.class is required" .Values.ingress.class | quote }}
   {{- if .Values.ingress.tls }}
   tls:
   {{- range .Values.ingress.hosts }}
   {{- if .tls }}
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/ingestion/templates/ingestion-ingress.yaml:26]: The readme is for TLS 
in deployment.md
```
helm package charts/ingestion/ -u && \
helm install ingestion-v0.1.0-dev ingestion-v0.1.0.tgz \
     --set image.tag=0.1.0 \
     --set image.repository=ingestion \
     --set dockerregistry=$ACR_SERVER \
     --set identity.clientid=$INGESTION_ID_CLIENT_ID \
     --set identity.serviceAccountName=ingestion-sa-v0.1.0 \
     --set identity.tenantId=$TENANT_ID \
     --set keyVaultName=$INGESTION_KEYVAULT_NAME \
     --set ingress.hosts[0].name=$EXTERNAL_INGEST_FQDN \
     --set ingress.hosts[0].serviceName=ingestion \
     --set ingress.hosts[0].tls=true \
     --set ingress.hosts[0].tlsSecretName=$INGRESS_TLS_SECRET_NAME \
     **--set ingress.tls=true \**
     --set ingress.class=nginx \
     --set secrets.queue.keyname=IngestionServiceAccessKey \
     --set secrets.queue.name=${INGESTION_QUEUE_NAME} \
     --set secrets.queue.namespace=${INGESTION_QUEUE_NAMESPACE} \
     --set reason="Initial deployment" \
     --set tags.dev=true \
     --namespace backend-dev \
     --dependency-update
```

File: charts/workflow/values.yaml [hunk 4]
```diff
@@ -5,16 +5,17 @@ dockerregistrynamespace:
...
 image:
   repository:
   tag:
   pullPolicy: IfNotPresent
 reason: unknown
 serviceuri:
-  delivery: http://delivery/api/Deliveries/
-  drone: http://dronescheduler/api/DroneDeliveries/
-  package: http://package/api/packages/
+  delivery: http://delivery-v010/api/Deliveries/
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/workflow/values.yaml:16]: A) It is not possible, you don't know which version like to work each dependency
For example, v010 delivery, v251 package, and done v100, it is a perfect combination. 
B) I can move as parameter, this is a parameter with default value, you can overwrite passing on command line.  I will move to the readme file.

File: charts/ingestion/templates/ingestion-ingress.yaml [hunk 5]
```diff
@@ -15,24 +15,26 @@ kind: Ingress
...
+    secretName: {{ .tlsSecretName }}
   {{- end }}
   {{- end }}
   {{- end }}
   rules:
   {{- range .Values.ingress.hosts }}
   - host: {{ .name }}
     http:
-      paths:
+      paths:
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/ingestion/templates/ingestion-ingress.yaml]: done

File: charts/workflow/values.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/workflow/values.yaml]: done

File: charts/delivery/templates/delivery-ingress.yaml [hunk 6]
```diff
@@ -15,16 +15,18 @@ kind: Ingress
...
 spec:
+  ingressClassName: {{ required "ingress.class is required" .Values.ingress.class | quote }}
   {{- if .Values.ingress.tls }}
   tls:
   {{- range .Values.ingress.hosts }}
   {{- if .tls }}
   - hosts:
     - {{ .name }}
-    secretName: {{ $relname }}-{{ .tlsSecretName }}
+    secretName: {{ .tlsSecretName }}
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/delivery/templates/delivery-ingress.yaml:29]: No, if you don't like. 
tlsSecretName  is a deploy parameter, you could  use the name you like for your version. You could use the same or not.
```
helm package charts/ingestion/ -u && \
helm install ingestion-v0.1.0-dev ingestion-v0.1.0.tgz \
     --set image.tag=0.1.0 \
     --set image.repository=ingestion \
     --set dockerregistry=$ACR_SERVER \
     --set identity.clientid=$INGESTION_ID_CLIENT_ID \
     --set identity.serviceAccountName=ingestion-sa-v0.1.0 \
     --set identity.tenantId=$TENANT_ID \
     --set keyVaultName=$INGESTION_KEYVAULT_NAME \
     --set ingress.hosts[0].name=$EXTERNAL_INGEST_FQDN \
     --set ingress.hosts[0].serviceName=ingestion \
     --set ingress.hosts[0].tls=true \
     **--set ingress.hosts[0].tlsSecretName=ingestion-ingress-tls \**
     --set ingress.tls=true \
     --set ingress.class=nginx \
     --set secrets.queue.keyname=IngestionServiceAccessKey \
     --set secrets.queue.name=${INGESTION_QUEUE_NAME} \
     --set secrets.queue.namespace=${INGESTION_QUEUE_NAMESPACE} \
     --set reason="Initial deployment" \
     --set tags.dev=true \
     --namespace backend-dev \
     --dependency-update
``` 
And, the SecretProviderClass is versioning, and you can use the same name in a different key vault. 
```

###################################################################################################
# Ingestion Secret Provider Class - CSI Driver
###################################################################################################
{{- $svcversion := .Chart.AppVersion | replace "." "" }}
apiVersion: secrets-store.csi.x-k8s.io/v1
kind: SecretProviderClass
metadata:
  name: ingestion-secrets-csi-akv-{{ $svcversion }}
  namespace: backend-dev
spec:
  provider: azure
  secretObjects:
  **- secretName: ingestion-ingress-tls**
    type: Opaque
    data:
    - objectName: Ingestion-Ingress-Tls-Key
      key: tls.key
    - objectName: Ingestion-Ingress-Tls-Crt
      key: tls.crt
  - secretName: ingestion-secrets
    type: Opaque
    data:
    - objectName: Queue--Key
      key: queue-keyvalue
    - objectName: ApplicationInsights--InstrumentationKey
      key: appinsights-ikey
  parameters:
    usePodIdentity: "false"
    **clientID: {{ .Values.identity.clientid }}
    keyvaultName: {{ .Values.keyVaultName }}**
    objects:  |
      array:
        - |
          objectName: Queue--Key
          objectAlias: Queue--Key
          objectType: secret
        - |
          objectName: ApplicationInsights--InstrumentationKey
          objectAlias: ApplicationInsights--InstrumentationKey
          objectType: secret
        - |
          objectName: Ingestion-Ingress-Tls-Key
          objectAlias: Ingestion-Ingress-Tls-Key
          objectType: secret
        - |
          objectName: Ingestion-Ingress-Tls-Crt
          objectAlias: Ingestion-Ingress-Tls-Crt
          objectType: secret
    tenantId: {{ .Values.identity.tenantId }}
---
```

File: charts/dronescheduler/templates/dronescheduler-identity.yaml [hunk 7]
```diff
@@ -6,28 +6,10 @@
 ###################################################################################################
 # Dronescheduler service identity
 ###################################################################################################
-{{- $fullname := include "dronescheduler.fullname" . | replace "." "" }}
-apiVersion: "aadpodidentity.k8s.io/v1"
-kind: AzureIdentity
+apiVersion: v1
+kind: ServiceAccount
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/dronescheduler/templates/dronescheduler-identity.yaml]: done

File: charts/ingestion/templates/ingestion-secret-provider.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    - objectName: ApplicationInsights--InstrumentationKey
+      key: appinsights-ikey
+  parameters:
+    usePodIdentity: "false"
+    clientID: {{ .Values.identity.clientid }}
+    keyvaultName: {{ .Values.keyVaultName }}
+    objects:  |
+      array:
+        - |
+          objectName: Queue--Key
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply charts/ingestion/templates/ingestion-secret-provider.yaml:39]: I created a task for the future. My idea is as soon as possible have a version working again, and to do improvement later on.

File: azuredeploy.bicep [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+            keyData: sshRSAPublicKey
+          }
+        ]
+      }
+    }
+    servicePrincipalProfile: {
+      clientId: servicePrincipalClientId
+      secret: servicePrincipalClientSecret
+    }
+    addonProfiles: {
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply azuredeploy.bicep:131]: I created a task for the future. My idea is as soon as possible have a version working again, and to do improvement later on.

File: azuredeploy.bicep [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+      {
+        name: 'agentpool'
+        osDiskSizeGB: osDiskSizeGB
+        count: agentCount
+        vmSize: agentVMSize
+        osType: osType
+        mode: 'System'
+      }
+    ]
+    linuxProfile: {
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply azuredeploy.bicep]: done

File: azuredeploy.bicep (hunk 9 above)
[the remark this answers is not in the document]
Comment (v-fearam) [reply azuredeploy.bicep:115]: added

**Review by ferantivero — APPROVED** [review]
LGTM

**Pull request merged**